libraries/botbuilder/tests/inspectionMiddleware.test.js

```diff
@@ -291,6 +290,272 @@ describe('InspectionMiddleware', function() {
+        // set up our expectations in nock - each corresponds to a trace message we expect to receive in the emulator
+
+        nock('https://test.com')
+            .post('/v3/conversations/Convo1/activities', activity => activity.type === 'trace'
+                && activity.value.text == 'hi')
+            .reply(200, { id: 'test' });
+
+        nock('https://test.com')
+            .post('/v3/conversations/Convo1/activities', activity => activity.type === 'trace'
+                && activity.value.text == 'echo: hi')
+            .reply(200, { id: 'test' });
+
+        nock('https://test.com')
+            .post('/v3/conversations/Convo1/activities', activity => activity.type === 'trace'
+                && activity.value.userState && activity.value.userState.x.property == 'hello'
+                && activity.value.conversationState && activity.value.conversationState.y.property == 'world')
+            .reply(200, { id: 'test' });
+
+        // create the various storage and middleware objects we will be using
+
+        const storage = new MemoryStorage();
+        const inspectionState = new InspectionState(storage);
+        const userState = new UserState(storage);
+        const conversationState = new ConversationState(storage);
+        const inspectionMiddleware = new InspectionMiddleware(inspectionState, userState, conversationState);
+
+        // the emulator sends an /INSPECT open command - we can use another adapter here
+
+        const openActivity = MessageFactory.text('/INSPECT open');
+
+        const inspectionAdapter = new TestAdapter(async (turnContext) => {
+            await inspectionMiddleware.processCommand(turnContext);
+        }, null, true);
+
+        await inspectionAdapter.receiveActivity(openActivity);
+
+        const inspectionOpenResultActivity = inspectionAdapter.activityBuffer[0];
+        const attachCommand = inspectionOpenResultActivity.value;
+
+        const adapter = new TestAdapter(async (turnContext) => {
+            try {
+                await inspectionMiddleware.onTurn(turnContext, null)
+                throw new Error('should have thrown an error.')
+            } catch (error) {
+                assert.strictEqual(error.message, 'next is not a function', 'next function should be null')
+            }
+        }, null, true);
+
+        adapter.use(inspectionMiddleware);
+
+        await adapter.receiveActivity(MessageFactory.text(attachCommand));
+
+        await adapter.receiveActivity();
+    });
 
+    it('should invokeInbound throw an error', async () => {
+        let warnExecuted = false;
+        const storage = new MemoryStorage();
+        const inspectionState = new InspectionState(storage);
+        const userState = new UserState(storage);
+        const conversationState = new ConversationState(storage);
+        const inspectionMiddleware = new InspectionMiddleware(inspectionState, userState, conversationState);
+
+        const warn = console.warn;
```

Comment:
Please add some comments to this test explaining the use of console.warn

Reply:
Hi @EricDahlvang , we use this to override the current warn behavior so it can be intercepted when it's being called. We added this comment to code also.

Reply:
@denscollo please use sinon instead of changing the builtin `console.warn`.

[Example import from botFrameworkAdapter.test.js](https://github.com/microsoft/botbuilder-js/blob/bc95461422f3b3ad74bc0374db7e70e8c6c72a9f/libraries/botbuilder/tests/botFrameworkAdapter.test.js#L12)